lib/Service/DelegationService.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/**
+ * @author Cyrille Bollu <cyr.debian@bollu.be>
+ *
+ * GroupFolders
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\GroupFolders\Service;
+
+use OCP\IConfig;
+use OCP\IGroup;
+use OCP\IGroupManager;
+use OCP\IRequest;
+use OCP\IUserManager;
+use OCP\IUserSession;
+
+class DelegationService {
+
+	/** @var IConfig */
+        private $config;
+
+        /** @var IGroupManager */
+        private $groupManager;
+
+        /** @var IUserSession */
+        private $userSession;
+
+	public function __construct(IConfig $config,
+                        IGroupManager $groupManager,
+			IUserSession $userSession) {
+                $this->config = $config;
+                $this->groupManager = $groupManager;
+		$this->userSession = $userSession;
+	}
+
+	/**
+	 * Return true if user is member of a group that has been granted admin rights on groupfolders
+	 *
+	 * @return bool
+	 */
+	public function isAdmin() {
+		$allowedGroups = explode('|', $config = $this->config->getAppValue('groupfolders', 'delegated-admins', 'admin'));
```

Comment:
This should also always return true if the user is a Nextcloud admin.

Reply:
done